### ISSUE TITLE
[PATCH] native_midi_macosx: support SDL_SOUNDFONTS

### ISSUE THREAD

**Issue by SDLBugzilla** [libsdl-org/SDL_mixer#290]
# This bug report was migrated from our old Bugzilla tracker.

These attachments are available in the static archive:

* [native_midi_macosx: support SDL_SOUNDFONTS (native_midi_macosx_set_soundfont.patch, text/plain, 2020-11-24 22:07:14 +0000, 3917 bytes)](https://bugzilla.libsdl.org/attachment.cgi?id=4542)

**Reported in version:** unspecified
**Reported for operating system, platform:** Mac OS X (All), All

# Comments on the original bug report:

On 2020-11-24 22:07:14 +0000,  wrote:

> Created attachment 4542
> native_midi_macosx: support SDL_SOUNDFONTS
> 
> Adds support for setting soundfont via SDL_SOUNDFONTS and friends in native_midi_macosx backend.

On 2020-12-01 21:49:22 +0000, Sam Lantinga wrote:

> Fixed, thanks!
> https://hg.libsdl.org/SDL_mixer/rev/3189c67fb0d9

**Comment by sezero** [comment]
The FIXME not at line 156 is scary: https://github.com/libsdl-org/SDL_mixer/blob/master/src/codecs/native_midi/native_midi_macosx.c#L156
@slouken, @icculus: what to do about it?

**Comment by sezero** [comment]
> The FIXME note at line 156 is scary: https://github.com/libsdl-org/SDL_mixer/blob/master/src/codecs/native_midi/native_midi_macosx.c#L156 @slouken, @icculus: what to do about it?

PING?

**Comment by a-hurst** [comment]
@sezero From how I'm reading the code, the issue is that if setting a soundfont fails and a) there's no default soundfont to revert to, or b) trying to set the default soundfont fails, the audio unit might not work somehow.

Based on [this blogpost](https://billthefarmer.github.io/blog/swift/), it looks like you can retrieve the error string if there's a failure on that function, so a solution might be to raise an SDLError with the relevant error text if the block with that comment gets hit (or even before, since it would presumably be helpful to know why a given soundfont failed to load before attempting to reload a fallback).

**Comment by slouken** [comment]
I agree this is scary, but I'm not sure what can be done. @icculus, any ideas?
